sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -112,3 +116,107 @@ case class ExplainCommand(
     ("Error occurred during query planning: \n" + cause.getMessage).split("\n").map(Row(_))
   }
 }
+
+/**
+ * A command that lists column names of a given table.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW COLUMNS (FROM | IN) table_identifier [(FROM | IN) database];
+ * }}}
+ */
+case class ShowColumnsCommand(table: TableIdentifier) extends RunnableCommand {
+  // The result of SHOW COLUMNS has one column called 'result'
+  override val output: Seq[Attribute] = {
+    AttributeReference("result", StringType, nullable = false)() :: Nil
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    sparkSession.sessionState.catalog.getTableMetadata(table).schema.map(c => Row(c.name))
+  }
+}
+
+/**
+ * A command that lists the partition names of a given table. If the partition spec is specified,
+ * partitions that match the spec are returned. [[AnalysisException]] is thrown under the following
+ * conditions:
+ *
+ *  1. If the command is called for a non partitioned table.
+ *  2. If the partition spec refers to the columns that are not defined as partitioning columns.
+ *
+ * This function creates a [[ShowPartitionsCommand]] logical plan
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW PARTITIONS [db_name.]table_name [PARTITION(partition_spec)]
+ * }}}
+ */
+case class ShowPartitionsCommand(
+    table: TableIdentifier,
+    spec: Option[TablePartitionSpec]) extends RunnableCommand {
+  // The result of SHOW PARTITIONS has one column called 'result'
+  override val output: Seq[Attribute] = {
+    AttributeReference("result", StringType, nullable = false)() :: Nil
+  }
+
+  private def getPartName(spec: TablePartitionSpec, partitionColumns: Seq[String]): String = {
+    partitionColumns.map { col =>
+      PartitioningUtils.escapePathName(col) + "=" + PartitioningUtils.escapePathName(spec(col))
+    }.mkString("/")
```

Comment:
@yhuai I was trying to remember why i changed it as i had it use "/" before :-). I think i looked at the function makePartName in Warehouse.java in hive and had decided to change the code. Just FYI.